contributors/design-proposals/federated-deployment.md

```diff
@@ -0,0 +1,88 @@
+
+On the other hand FederatedReplicaSets should somehow reflect the status of the underlying cluster - like number of total replicas that are there so that kubectl rollout has proper data to work on. 
+
+<img src="federated-deployment-3.png" width="500"/>
+
+Federated nginx-701339712 gets all of the statistics as well as spec elements (like the total number of replicas) from ReplicaSets in the underlying clusters. Obviously none of the spec changes should be pushed back. With this change the whole thing would look like as a well-working “whole” while in fact Federated ReplicaSets would be “puppetized” - their original functionality (creating rs in clusters and controlling the balancing) would be turned off. 
+
+## Implementation
+
+The implementation of Federated Deployment Controller is largely based on the existing Federated ReplicaSet Controller. The following changes however will be added (on top of obvious changes like “s/replicaset/deployment/”):
+
++ For each of the pod templates, Federated Deployment will make sure that puppetized FederatedReplicaset with the name deployment_name + “-” + pod_template_hash is created, with the appropriate revision number.
+
++ Local deployment checks will occur according to annotation (if present) or alphabetically.
++ If Federated Deployment with `strategy == RollingUpdate` spots a Local Deployment with different pod template it updates its only if there is no other update going on. Updates are spotted by checking if:
+    + `deployment.status.updatedReplicas == deployment.status.replicas`
+    + `deployment.status.unavailableReplicas == 0`
+    + `deployment.status.observed_generation == deployment.metadata.generation`
+
+  The Local Deployment will get a chance to be updated at a later time. Every change in Local Deployments trigger a Federated Deployment reconciliation. So once one Local Deployment is done with the update and publishes its new state the reconciliation will be triggered and next Local Deployment update started.
++ Federated Deployments with `strategy == Recreate` update all clusters at once.
++ FederatedDeployments always update replica numbers (if needed), no matter whether some update is going on or not.
++ Rollback REST api is registered.
++ Federated Deployment Controller has support for deleting old Federated ReplicaSets (just like in regular deployments). Replicasets are deleted in a non-cascading way.
+
+Moreover a “puppetized” Federated ReplicaSet (PFRS) support will be added to Federated Replicaset Controller:
+
+ + PFRS are marked with an annotation. The presence of the annotation makes the FRS a puppet.
+ + PFRS are created by Federated Deployments and contain the proper pod spec.
+ + PFRS monitors all underlying clusters where the replicas could potentially be found and updates their statuses and spec. Replicas based on the underlying cluster content.
```

Comment:
To clarify, are you saying that it will only update status.Replicas from underlying clusters or the whole status? Updating whole status seems fine to me